src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -497,8 +497,14 @@ private static bool RemoveEmptyDirectory(string fullPath, bool topLevel = false,
                     case Interop.Error.EACCES:
                     case Interop.Error.EPERM:
                     case Interop.Error.EROFS:
+                        // Prefer throwing DirectoryNotFoundException over UnauthorizedAccess IOException.
+                        if (topLevel && !DirectoryExists(fullPath, out Interop.ErrorInfo existErr) && existErr.Error == Interop.Error.ENOENT)
+                        {
+                            throw Interop.GetExceptionForIoErrno(Interop.Error.ENOENT.Info(), fullPath, isDirectory: true);
+                        }
+                        throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, fullPath));
```

Comment:
nit, if you want you could do 
```suggestion
                        goto case Interop.Error.EISDIR;
```

Reply:
I think it should be safe to remove `EISDIR` here. It is not an error that `rmdir` should return, since it is meant to be used with directories.